app/models/bank_account.rb

```diff
@@ -0,0 +1,28 @@
+require 'roo'
+
+class BankAccount < ActiveRecord::Base
+
+  has_many :bank_transactions, dependent: :destroy
+
+  normalize_attributes :name, :iban, :description
+
+  validates :name, :presence => true, :uniqueness => true, :length => { :minimum => 2 }
+  validates :iban, :presence => true, :uniqueness => true
+  validates_format_of :iban, :with => /\A[A-Z]{2}[0-9]{2}[0-9A-Z]{,30}\z/
```

Comment:
What about a full IBAN check with one of the gems (like [iban-tools](https://github.com/alphasights/iban-tools))?
I've seen users in the USA or Israël, do we need to think about them regarding IBAN? I guess being able to have it not (so) visible would be enough (and given enough interest, they could submit a PR to generalise is).

Reply:
it's the same format we accept also at user and supplier. iban-tools also accepts ibans with spaces, witch is a problem when comparing them. IMHO the basic format is enough for now, so people could also insert "invalid IBANs" for countries without defined format